commons/ava_networks/high_level_gecko_client.go

```diff
@@ -40,6 +43,46 @@ func NewGeckoUser(username string, password string) *GeckoUser {
 }
 
 
+func (highLevelGeckoClient HighLevelGeckoClient) AddValidatorOnSubnet(
+	nodeId string,
+	pchainAddress string,
+	stakeAmount int64) error {
+	client := highLevelGeckoClient.client
+	currentPayerNonce, err := highLevelGeckoClient.getCurrentPayerNonce(pchainAddress)
+	if err != nil {
+		return stacktrace.Propagate(err, "Failed to get payer nonce from address %s", pchainAddress)
+	}
+	stakingStartTime := time.Now().Add(TIME_UNTIL_STAKING_BEGINS).Unix()
+	addStakerUnsignedTxn, err := client.PChainApi().AddDefaultSubnetValidator(
+		nodeId,
+		stakingStartTime,
+		time.Now().Add(TIME_UNTIL_STAKING_ENDS).Unix(),
+		stakeAmount,
+		currentPayerNonce + 1,
+		pchainAddress,
+		DELEGATION_FEE_RATE)
+	if err != nil {
+		return stacktrace.Propagate(err, "Failed to add default subnet delegator %s", nodeId)
+	}
+	addStakerSignedTxn, err := client.PChainApi().Sign(
+		addStakerUnsignedTxn,
+		pchainAddress,
+		highLevelGeckoClient.geckoUser.username,
+		highLevelGeckoClient.geckoUser.password)
+	if err != nil {
+		return stacktrace.Propagate(err, "Failed to sign staker transaction.")
+	}
+	_, err = client.PChainApi().IssueTx(addStakerSignedTxn)
+	if err != nil {
+		return stacktrace.Propagate(err, "Failed to issue staker transaction.")
+	}
+	for time.Now().Unix() < stakingStartTime {
```

Comment:
Ouch, this means we'll have a _30 second_ sleep. Is this absolutely necessary? This will likely slow everything down a buuunch

Reply:
@mieubrisse  I'll see how low I can get this 👍

Reply:
20 seconds is the lowest i can get it to run reliably locally